server/src/providers/regression-model.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@nestjs/common';
+const MLR = require('ml-regression-multivariate-linear');
+import * as fs from 'fs';
+@Injectable()
+export class RegressionModel {
+    private model: any;
+    private logsEnabled: boolean;
+    private inputs: any;
+    private outputs: any;
+    constructor(logsEnabled= false) {
+        this.logsEnabled = logsEnabled;
+        this.log('Logs are enabled');
+    }
+    // Trains the model with the inputs
+    public trainModel(inputs, output) {
+        try {
+            this.model = new MLR(inputs, output);
+            this.log('Model has been trained');
+        } catch (err){
+            throw err;
+        }
+    }
+
+    // Prediction of the model
+    public predict(inputs) {
+        try {
+            const predictions = [];
```

Comment:
```suggestion
            const predictions = inputs.map(input => this.model.predict(input));
```

Then you can remove the .forEach code block